### DISCUSSION TITLE
How to derive environment variables for EdgeX services ?

### DISCUSSION THREAD

**Discussion by M0hanrajp** [edgexfoundry/community#297] [Q&A] [answered]
**Question:**

How to Identify environment variables for an X Edgex service ? what is the complete scope of the variables ?

**Description:**

There is a documentation in edgex which mentions [Environment Variables](https://docs.edgexfoundry.org/3.1/microservices/configuration/CommonEnvironmentVariables/) Here there are few common environment variables mentioned.

What I'm looking for is a source material that allows me to write or include these variables in my docker compose file for edgex services.

For example:

If I want to add few variables regarding rules engine I would follow [Basic configurations](https://ekuiper.org/docs/en/latest/configuration/global_configurations.html)

![image](https://github.com/user-attachments/assets/fcf05a77-0b7e-4073-b3ce-934cd8378a75)

Here it's clear that what is the format of the environment variable and what does the variable do.

**So similarly I wanted to know what are the other variables that are available for me to use in edgex ?** 

**I want to learn what variables are used for exporting data to cloud, if we can make a service subscribe to some other topic other than the default one ? please help here.**

Thanks.

**Comment by cloudxxx8** [comment, marked as answer]
You have found the standard EdgeX enviromnet variable docs, and it describes all of them.
For the individual service, the envorionment variables are for the configuration override
https://docs.edgexfoundry.org/3.1/microservices/configuration/CommonEnvironmentVariables/#service-configuration-overrides

You can check each service configuration from their individual doc page, for example, https://docs.edgexfoundry.org/3.1/microservices/core/metadata/Configuration/

Or, you can visit the source directly
https://github.com/edgexfoundry/edgex-go/blob/napa/cmd/core-metadata/res/configuration.yaml
